generator/templates/server/server.gotmpl

```diff
@@ -562,3 +596,37 @@ func (s *Server) TLSListener() (net.Listener, error) {
 	}
 	return s.httpsServerL, nil
 }
+
+func (s *Server) interruptChan() chan os.Signal {
+    s.chanLock.Lock()
+    defer s.chanLock.Unlock()
+
+    if s.interrupt == nil {
+        s.interrupt = make(chan os.Signal, 1)
+    }
+    
+    return s.interrupt
+
+}
+
+func (s *Server) handleInterrupt(interrupt chan os.Signal, quitting chan struct{}, l net.Listener) {
+    for _ = range interrupt {
+        if s.interrupted {
+            s.Logf("Server already shutting down")
+            continue
+        }
+        s.interrupted = true
```

Comment:
these 4 would be better if they used an atomic.
This entire method can be run in a sync.Once actually

Reply:
Only niladic functions can run in sync.Once - and I don't think they can have receivers, i.e. be methods.

Reply:
it's just a wrapper around the entire block of code, you have to lift the variables you want to capture out of there by declaring those outside of the closure